Src/SDK/Common/Temporal.Common.Payloads/internal/PayloadContainersEnumerableJsonConverter.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using JsonSerializer = Newtonsoft.Json.JsonSerializer;
+
+namespace Temporal.Common.Payloads
+{
+    public class PayloadContainersEnumerableJsonConverter : JsonConverter<PayloadContainers.Enumerable>
```

Comment:
My unit test failed until I wrote the custom converter. The data wouldn't serialize. I'm glad you brought it up because I don't want it to be necessary, so the problem could be the assumptions I made when I wrote the test.

Reply:
Got it. We can discuss and potentially investigate together.

Reply:
Found the underlying issue and resolved the issue by adding proper treatment for JObject.